From a844fa0ba0eccd7d16f63578417bec4f7244ca25 Mon Sep 17 00:00:00 2001
From: Thomas Lamprecht <t.lamprecht@proxmox.com>
Date: Wed, 22 Sep 2021 16:25:07 +0200
Subject: [PATCH backup] move dump-catalog-shell-cli doc-helper to
 proxmox-backup-client crate

it's only used for generating the docs for the interactive-shell
parts of the client.

Ideally we'd avoid that whole separate binary in the first place and
let the client dump it, but we'd need to have some more elaborate
"hide this command from the help/usage" mechanisms in the CLI
helper/formatter code to make that play out more nicely.

Signed-off-by: Thomas Lamprecht <t.lamprecht@proxmox.com>
---
 Makefile                                                        | 2 +-
 .../src}/bin/dump-catalog-shell-cli.rs                          | 0
 2 files changed, 1 insertion(+), 1 deletion(-)
 rename {src => proxmox-backup-client/src}/bin/dump-catalog-shell-cli.rs (100%)

diff --git a/Makefile b/Makefile
index 90723241..f2f2363b 100644
--- a/Makefile
+++ b/Makefile
@@ -179,6 +179,7 @@ $(COMPILED_BINS) $(COMPILEDIR)/dump-catalog-shell-cli $(COMPILEDIR)/docgen: .do-
 	    --bin proxmox-backup-banner \
 	    --package proxmox-backup-client \
 	    --bin proxmox-backup-client \
+	    --bin dump-catalog-shell-cli \
 	    --bin proxmox-backup-debug \
 	    --package proxmox-file-restore \
 	    --bin proxmox-file-restore \
@@ -190,7 +191,6 @@ $(COMPILED_BINS) $(COMPILEDIR)/dump-catalog-shell-cli $(COMPILEDIR)/docgen: .do-
 	    --package proxmox-restore-daemon \
 	    --bin proxmox-restore-daemon \
 	    --package proxmox-backup \
-	    --bin dump-catalog-shell-cli \
 	    --bin proxmox-daily-update \
 	    --bin proxmox-file-restore \
 	    --bin proxmox-tape \
diff --git a/src/bin/dump-catalog-shell-cli.rs b/proxmox-backup-client/src/bin/dump-catalog-shell-cli.rs
similarity index 100%
rename from src/bin/dump-catalog-shell-cli.rs
rename to proxmox-backup-client/src/bin/dump-catalog-shell-cli.rs
